python_modules/dagster/dagster/_core/definitions/declarative_automation/operands/operands.py

```diff
@@ -55,10 +57,19 @@ def _is_initial_evaluation(self, context: AutomationContext) -> bool:
         if previous_requested_subset is None:
             return True
 
-        previous_subset_value_type = type(previous_requested_subset.get_internal_value())
+        previous_subset_value = previous_requested_subset.get_internal_value()
+        previous_subset_value_type = type(previous_subset_value)
 
         current_subset = context.asset_graph_view.get_empty_subset(key=context.root_context.key)
         current_subset_value_type = type(current_subset.get_internal_value())
+        # if we have a key ranges subset, we can compare the partitions snapshot
+        if isinstance(previous_subset_value, KeyRangesPartitionsSubset):
```

Comment:
any rollback safety concerns here? If there were, we could push the code that is able to handle KeyRangesPartitionsSubset on the read path first, then separately push out the change to actually add them to the relevant subsets

Reply:
yeah that's a good point, disabling the write path in this PR, stacking a pr to enable it on top